Singleplayer-Tutorial

alptuncs opened the pull request:
- [x] Hedefleri rastgele yerleştirip oyunu başlatma
- [x] Oyuncudan input alıp o koordinatı deneme ve tahtayı o koordinat vuruldu olarak güncelleme
- [x] Input kontrolü
- [x] Oyun bitme koşulları gerçekleşmediği sürece oyunu devam ettirme
- [x] Oyuncunun performansına göre bir skor sistemi
- [x] Tüm gemiler vurulunca oyunu bitirme

alptuncs committed:
initial commit

alptuncs committed:
Game Manager

Game Manager eklendi, oyun tutorial şeklinde Lives veya gemiler bitene kadar çalışıyor, input kontrolü de yapılıyor

alptuncs committed:
shipValue

shipValue yerleştirilen gemilere göre hesaplanıyor artık sabit değil, tüm gemiler vurulunca oyun bitiyor

alptuncs committed:
Input Check

input check düzeltildi

SeferMirza commented on Battleships/Battleships/GameManager.cs:
```diff
@@ -12,22 +13,115 @@ public class GameManager
             _computerBoard = computerBoard;
             _boardRenderer = boardRenderer;
+
+            Target amiralGemisi = new Target(4, "east", "amiralGemisi");
+            Target kruvazor = new Target(3, "north", "kruvazor");
+            Target mayinGemisi = new Target(2, "east", "mayinGemisi");
+            Target denizalti = new Target(1, "north", "denizalti");
+
+            targets.Add(amiralGemisi);
+            targets.Add(kruvazor);
+            targets.Add(mayinGemisi);
+            targets.Add(denizalti);
```
Yeni yaratma işlerini burada yapma buraya parametre ile gelsin 
Factory design patterni bi incele istersen

alptuncs replied:
Değiştirdim, bir doğru koordinat verince hata alıyordum onları düzelttim şuan çalışıyor.

SeferMirza replied:
commit attın mı ?

alptuncs replied:
attım şimdi

alptuncs committed:
Target List

cihandeniz commented on Battleships/Battleships/GameManager.cs:
```diff
@@ -12,22 +13,107 @@ public class GameManager
+                shipValue += target.Size * target.Size * (5 - target.Size);
+            }
+        }
+        public void play()
```
method'lar buyuk harfle basliyor

alptuncs replied:
düzelttim

cihandeniz commented on Battleships/Battleships/GameManager.cs:
```diff
@@ -12,22 +13,107 @@ public class GameManager
             _boardRenderer = boardRenderer;
+            _targets = targets;
+
+            foreach (Target target in _targets)
+            {
+                computerBoard.RandomPlaceShip(5 - target.Size, target);
+                shipValue += target.Size * target.Size * (5 - target.Size);
+            }
```
bu tip initialization'lari constructor'da yapmayin derler. bunu ayir istersen, builder pattern kullanirsin yaratan yerde. `new GameManager(..).Init()` filan gibi

alptuncs replied:
ComputerBoard'a  gemi yerleştirmeyi ayrı bir fonksiyona aldım

cihandeniz commented on Battleships/Battleships/GameManager.cs:
```diff
@@ -12,22 +13,107 @@ public class GameManager
+            RenderGame();
         }
-        public void FireMissile(BoardManager board, int i, int j)
+        private void FireMissile(BoardManager board, int i, int j)
```
i ve j hep beraber geziyorsa, Coordinate diye bir class yap, ya da dotnet'te varsa onu kullan.

alptuncs replied:
Coordinate class'ı oluşturdum diğer i ve j veya x ve y alan fonksiyonları Coordinate alıcak şekilde güncelledim

alptuncs committed:
Coordinate Class

Coordinate class eklendi Board manager, Game manager ve Board Renderer method'ları güncellendi, method isimledi düzeltildi, computer board initialize edilmesi ayrı bir method'a alındı

alptuncs committed:
Initialize

cihandeniz commented on Battleships/Battleships/Coordinate.cs:
```diff
@@ -0,0 +1,21 @@
+{
+    public class Coordinate
+    {
+        private int _Xpos;
+        private int _Ypos;
```
field isimleri küçük harf

alptuncs replied:
düzelttim

cihandeniz commented on Battleships/Battleships/Coordinate.cs:
```diff
@@ -0,0 +1,21 @@
+        private int _Ypos;
+        public int XPos => _Xpos;
+        public int YPos => _Ypos;
+        public Coordinate(int i, int j)
```
xPos, yPos olmalı, i ve j yerine

alptuncs replied:
Düzelttim

cihandeniz commented on Battleships/Battleships/Coordinate.cs:
```diff
@@ -0,0 +1,21 @@
+    {
+        private int _Xpos;
+        private int _Ypos;
+        public int XPos => _Xpos;
+        public int YPos => _Ypos;
```
burada auto property kullanabilirsin, autoproperty'ler readonly olabiliyor, internette bir bak istersen, anlayacaksın dediğimi

alptuncs replied:
Readonly hale getirdim

cihandeniz replied:
olmamis bu, `public int XPos { get; private set; }` yapacaksin, field'i da sileceksin

cihandeniz commented on Battleships/Battleships/Program.cs:
```diff
@@ -12,33 +12,26 @@ internal class Program
         static void Main(string[] args)
         {
             IConsole console = new SystemConsole();
+            BoardRenderer boardRenderer = new BoardRenderer(10, 10);
+            BoardManager playerBoard = new BoardManager(10, 10);
+            BoardManager computerBoard = new BoardManager(10, 10);
+            List<Target> targets = new List<Target>();
             Target amiralGemisi = new Target(4, "east", "amiralGemisi");
             Target kruvazor = new Target(3, "north", "kruvazor");
             Target mayinGemisi = new Target(2, "east", "mayinGemisi");
             Target denizalti = new Target(1, "north", "denizalti");
```
c#'ta sol tarafta `var` kullanabiliyorsun, class adini cift yazmamis oluyorsun.

alptuncs replied:
Böyle olan yerlerde var kullandım

alptuncs committed:
Input Check update

alptuncs committed:
Unit Tests

alptuncs committed:
ReadOnly

alptuncs committed:
ReadOnly
